src/components/Tabs/tests/Tabs.test.tsx

```diff
@@ -193,4 +206,245 @@ describe('<Tabs />', () => {
+          trigger(tabs.find('ul'), 'onKeyUp', {
+            key: 'ArrowRight',
+          });
+        }
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+
+    describe('ArrowDown', () => {
+      it('shifts focus to the next tab when pressing ArrowDown', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowDown',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+
+      it('shifts focus to the first tab when pressing ArrowDown on the last tab', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        let x;
+        for (x = 0; x <= 3; x++) {
+          trigger(tabs.find('ul'), 'onKeyUp', {
+            key: 'ArrowDown',
+          });
+        }
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+
+    describe('ArrowLeft', () => {
+      it('shifts focus to the last tab when pressing ArrowLeft', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowLeft',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(2);
+      });
+
+      xit('shifts focus to the first tab when pressing ArrowLeft on the second tab', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowLeft',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+
+    describe('ArrowUp', () => {
+      it('shifts focus to the last tab when pressing ArrowUp', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} selected={0} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowUp',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(2);
+      });
+
+      it('shifts focus to the first tab when pressing ArrowUp on the second tab', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowLeft',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+  });
+
+  describe('getVisibleAndHiddenTabIndices()', () => {
```

Comment:
If this is a utility you probably wanna move this to its own test file. E.g. as folder structure:

```
utilities/
  index.ts
  getVisibleAndHiddenTabIndices.ts
  tests/
    getVisibleAndHiddenTabIndices.test.ts
```

Reply:
That would be inconsistent with the rest of our repo. If I did this, I would rather do them all at once.